Public/Src/Tools/Execution.Analyzer/Analyzers.Core/XLGPlusPlus/XldbDataStore.cs

```diff
@@ -68,14 +68,133 @@ public IEnumerable<string> GetEventsByType(ExecutionEventId eventTypeID)
                 {
                     foreach (var kvp in database.PrefixSearch(eventQuery.ToByteArray()))
                     {
-                        storedEvents.Add(BXLInvocationEvent.Parser.ParseFrom(kvp.Value).ToString());
+                        switch (eventTypeID)
+                        {
+                            case ExecutionEventId.FileArtifactContentDecided:
+                                storedEvents.Add(FileArtifactContentDecidedEvent.Parser.ParseFrom(kvp.Value).ToString());
+                                break;
+                            case ExecutionEventId.WorkerList:
+                                storedEvents.Add(WorkerListEvent.Parser.ParseFrom(kvp.Value).ToString());
+                                break;
+                            case ExecutionEventId.PipExecutionPerformance:
+                                storedEvents.Add(PipExecutionPerformanceEvent.Parser.ParseFrom(kvp.Value).ToString());
+                                break;
+                            case ExecutionEventId.DirectoryMembershipHashed:
+                                storedEvents.Add(DirectoryMembershipHashedEvent.Parser.ParseFrom(kvp.Value).ToString());
+                                break;
+                            case ExecutionEventId.ProcessExecutionMonitoringReported:
+                                storedEvents.Add(ProcessExecutionMonitoringReportedEvent.Parser.ParseFrom(kvp.Value).ToString());
+                                break;
+                            case ExecutionEventId.ProcessFingerprintComputation:
+                                storedEvents.Add(ProcessFingerprintComputationEvent.Parser.ParseFrom(kvp.Value).ToString());
+                                break;
+                            case ExecutionEventId.ExtraEventDataReported:
+                                storedEvents.Add(ExtraEventDataReported.Parser.ParseFrom(kvp.Value).ToString());
+                                break;
+                            case ExecutionEventId.DependencyViolationReported:
+                                storedEvents.Add(DependencyViolationReportedEvent.Parser.ParseFrom(kvp.Value).ToString());
+                                break;
+                            case ExecutionEventId.PipExecutionStepPerformanceReported:
+                                storedEvents.Add(PipExecutionStepPerformanceReportedEvent.Parser.ParseFrom(kvp.Value).ToString());
+                                break;
+                            case ExecutionEventId.PipCacheMiss:
+                                storedEvents.Add(PipCacheMissEvent.Parser.ParseFrom(kvp.Value).ToString());
+                                break;
+                            case ExecutionEventId.ResourceUsageReported:
+                                storedEvents.Add(StatusReportedEvent.Parser.ParseFrom(kvp.Value).ToString());

```

Comment:
The names don't fully line up, intentional?

Reply:
The names didn't line up in our BXL code as well for that enum, and I kept the same names for the protobuf enums. It is a bit confusing but I'm assuming we can't change it in the BXL codebase due to telemetry related reasons (and so I'm not sure if I should change it here either).

Reply:
As discussed offline, the mismatch will be kept to mirror what we have in the BXL codebase